src/App.tsx

```diff
@@ -1,7 +1,46 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State =
+{
+  pressedKey: string | null,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    pressedKey: null,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleDocumentClick);
+  }
+
+  componentWillUnmount(): void {
+    document.addEventListener('keyup', this.handleDocumentClick);
+  }
+
+  handleDocumentClick = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: event.key });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <p className="App__message">
+          {
+            !this.state.pressedKey
+              ? 'Nothing was pressed yet'
+              : `The last pressed key is [${this.state.pressedKey}]`
+          }
```

Comment:
- no need to add spacing here. The code doesn't get any clearer, but you lose in ergonomics a lot
- let's start conditional rendering with positive condition - it is easier to read
```suggestion
          {!this.state.pressedKey
              ? 'Nothing was pressed yet'
              : `The last pressed key is [${this.state.pressedKey}]`}
```